### DISCUSSION TITLE
Idiots guide to configuring Rev23 PrimeNet?

### DISCUSSION THREAD

**Discussion by SirNibble** [prirun/p50em#20] [General]
Rather than wade through 300+ pages of PrimeNet guides, is there a nice & quick guide to adding a new machine to the example Ring?

I've updated the ring.cfg 

```
1 127.0.0.1:9019 abcdefghijklmnop  # Rev 19
2 127.0.0.1:9120 bcdefghijklmnopq  # Rev 20 - EM20 (aka GWHB)
3 127.0.0.1:9021 cdefghijklmnopqr  # Rev 21
4 127.0.0.1:9022 defghijklmnopqrs  # Rev 22
5 127.0.0.1:9023 efghijklmnopqrst  # Rev 23
6 127.0.0.1:9023 fghijklmnopqrstu  # Rev 24
7 127.0.0.1:9018 ghijklmnopqrstuv  # Rev 18
8 127.0.0.1:9123 hijklmnopqrstuvw  # Rev 23 - PIP01 (Rev 23.3.0.R33)

```
Configuring this back in the day was above my paygrade.

**Comment by hashbackup** [comment]
To add a new node you have to:

1. create a network config on the new node with the node-ids and system names of all the other nodes.  This is rather painful: you have to create a ring plus 8 nodes, and for each node you have to configure which other nodes it can access, so N^2 edits.

2. update the Primenet config file on every node where you want the new node visible.  For R19, you run netcfg and it updates netcon in cmdnc0.  For later revs you run primenet*>cmdnc0>config_net and give it the name of the network config file, which is primenet*>primenet.config.

3. update start_net and remote disk system names in cmdnc0>primos.comi

Shutdown all running systems and make backup copies of the system disk image before starting this, and/or make copies of primenet*>primenet.config.  I don't think a network configuration error will get you into an unbootable situation but it's best to avoid the risk.  Good luck!

**Comment by SirNibble** [comment]
Thanks.

Looks like it could be a bit too much of a phaff adding to the current ring, so I'll back-burner networking my 'clean' install for now until I've time to properly understand the How & Why of configuring it.

EDIT TO ADD - Cheated for now. 

Least interrested (for now) in the R18 example, so on my 'personal' copy of the R20 example used config_net to rename EM18 to PIP01 (my clean 23.3.0.R33 install).
Rebooted. EM20 and 'stat net' showed EM20 up and PIP01 down.

MagSav'd 'primenet.config' on EM20 and restored on PIP01

Ran 'start_net primenet*>primenet.config -node pip01' on PIP01 and .... voila!

```
 stat net
Ring network
      Node      State
    PIP01       ****
    EM19        Down
    EM20         Up
    EM21        Down
    EM22        Down
    EM23        Down
    EM24        Down
```

Yeah, the rest of the example ring may complain, but it'll do for now until I can get round to learning, and doing it all, from scratch.
